### PR TITLE
Updated e2e tests for cluster autoscaling.

### PR DESCRIPTION
**Pull request by jszczepkowski**
[Analytics]()

Updated e2e tests for cluster autoscaling to work with the new cluster autoscaler.

### CODE REVIEW

**Comment by jszczepkowski** [comment]
CC @fgrzadkowski

File: cluster/saltbase/salt/cluster-autoscaler/cluster-autoscaler.manifest [hunk 1]
```diff
@@ -15,7 +15,7 @@
         "containers": [
             {
                 "name": "cluster-autoscaler",
-                "image": "gcr.io/mwielgus-proj/cluster-autoscaler:v0.0.1-alpha2-4",
```
Comment (piosz) [inline cluster/saltbase/salt/cluster-autoscaler/cluster-autoscaler.manifest]: Let's make official release ASAP.

cc @fgrzadkowski @mwielgus

File: test/e2e/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -29,15 +28,12 @@ import (
 )
 
 const (
-	scaleUpTimeout   = 20 * time.Minute
-	scaleDownTimeout = 30 * time.Minute
+	scaleTimeout = 2 * time.Minute
 )
 
 // [Feature:ClusterSizeAutoscaling]: Cluster size autoscaling is experimental
```
Comment (piosz) [inline test/e2e/cluster_size_autoscaling.go]: This is not true. The feature no longer requires GCM enabled, though we still want to run this test in the dedicated suite.

File: test/e2e/cluster_size_autoscaling.go [hunk 3]
```diff
@@ -56,117 +52,55 @@ var _ = framework.KubeDescribe("Cluster size autoscaling [Feature:ClusterSizeAut
...
-
-		framework.ExpectNoError(framework.WaitForClusterSize(f.Client, nodeCount, scaleDownTimeout))
-	})
-
-	It("Should scale cluster size based on cpu reservation", func() {
-		setUpAutoscaler("cpu/node_reservation", 0.5, nodeCount, nodeCount+1)
+	It("Should correctly handle pending pods", func() {
+		By("Too large pending pod does not increase cluster size")
+		ReserveMemory(f, "memory-reservation", 1, memCapacityMb, false)
+		// Verify, that cluster size is not chagned
```
Comment (piosz) [inline test/e2e/cluster_size_autoscaling.go]: typo changed

File: test/e2e/cluster_size_autoscaling.go [hunk 4]
```diff
@@ -56,117 +52,55 @@ var _ = framework.KubeDescribe("Cluster size autoscaling [Feature:ClusterSizeAut
...
+		time.Sleep(scaleTimeout)
```
Comment (piosz) [inline test/e2e/cluster_size_autoscaling.go]: Add TODO to figure out better way to test this

**Comment by piosz** [comment]
LGTM, just a few nits.

File: test/e2e/cluster_size_autoscaling.go (hunk 3 above)
Comment (jszczepkowski) [inline test/e2e/cluster_size_autoscaling.go]: done

File: test/e2e/cluster_size_autoscaling.go (hunk 4 above)
Comment (jszczepkowski) [inline test/e2e/cluster_size_autoscaling.go]: done

File: test/e2e/cluster_size_autoscaling.go (hunk 2 above)
Comment (jszczepkowski) [inline test/e2e/cluster_size_autoscaling.go]: done

**Comment by jszczepkowski** [comment]
PTAL

**Comment by piosz** [comment]
LGTM

**Pull request merged**